htdocs/takepos/index.php

```diff
@@ -824,7 +828,7 @@ function WeighingScale(){
 
 <body class="bodytakepos" style="overflow: hidden;">
 <?php
-$keyCodeForEnter = getDolGlobalInt('CASHDESK_READER_KEYCODE_FOR_ENTER'.$_SESSION['takeposterminal']) > 0 ? getDolGlobalInt('CASHDESK_READER_KEYCODE_FOR_ENTER'.$_SESSION['takeposterminal']) : '';
+$keyCodeForEnter = getDolGlobalInt('CASHDESK_READER_KEYCODE_FOR_ENTER'. isset($_SESSION["takeposterminal"]) ? $_SESSION["takeposterminal"] : '') > 0 ? getDolGlobalInt('CASHDESK_READER_KEYCODE_FOR_ENTER'. isset($_SESSION["takeposterminal"]) ? $_SESSION["takeposterminal"] : '') : '';
```

Comment:
To be sure the ? test is execute before the . (before the isset), you should add parenthesis.
getDolGlobalInt('CASHDESK_READER_KEYCODE_FOR_ENTER'. (isset($_SESSION["takeposterminal"]) ? $_SESSION["takeposterminal"] : '')) > 0